### ISSUE TITLE
Opentoonz internal Scanner menu (Deprecated?)

### ISSUE THREAD

**Issue by riverfeya** [opentoonz/opentoonz#3225]
How to reset scaner settings?
Where OT save scaner settings?
Im change scaner mode to  "inner" and cant cjoose it after that anymore (
OT are freeze when i try to choose "scaner settings"

**Comment by RodneyBaker** [comment]
According to the Opentoonz documentation Opentoonz does not support internal scanning.
Instead GTS is used.

Please see:  https://opentoonz.readthedocs.io/en/latest/scanning_paper_drawings.html

Note that the documentation itself is maintained for scanning but see the information at the top of the page.

I do think that if Opentoonz doesn't support internal scanning and there is no plan to resolve that fairly soon... those menu options should likely be removed.

**Comment by RodneyBaker** [comment]
I've added this report is added to the 2020 UI scrub mostly to initiate a review of Opentoonz internal scanner options and to begin to chart a path forward for update or deprecation of scanner options in the menu.  

The following presses off topic of the current report but in reviewing scanner options it is seen that Scanning and Cleanup as a menu category could use some attention.  In the meantime I don't what to lose the topic of discussion so... @riverfeya any information you can provide will help us steer this back on topic concerning Scanner Settings.


There are reports that some have used the internal scanner options but this does not sync with documentation which reports those options no longer work, therefore recommending use of the external GTS application for scanning purpose.  My own personal experience is that I have never been able to use Opentoonz scanning.

For Opentoonz documentation of scanning see:  https://opentoonz.readthedocs.io/en/latest/scanning_paper_drawings.html

One way forward to rectify the 'scanning' part of 'Scanning and Cleanup' might be to reorganize the Scanning and Cleanup menu to better allow for a multi tier approach to ingesting imagery into Opentoonz.  Ingest operations include:

- Import (see also:  Open/Load/Import/Replace/Reload)
-- Direct ingest with/without conversion
-- Direct ingest with/without cleanup
-- Direct ingest with/without color separation

**- Scanning
-- Direct scan (initiate scanning process)
-- GTS scan (initiate GTS application)**

- Cleanup
-- Internal processes that focus on cleanup of ingested imagery
-- Internal processes that focus on conversion of ingested imagery

- Pencil Test/Stop Motion
-- There are now the Pencil Test and Stop Motion Control options of ingest that provide a new and important element to the 'Scanning and Cleanup' category.

- Other/Misc Methods of ingest (and bypass methodologies)
-- Browser ingest
-- Batch Cleanup
-- Copy/Paste conversions
-- Command line processing
-- Batch Rendering
-- Command line rendering

In short, the entire 'Scanning and

**Comment by gab3d** [comment]
It'd be good to revive the scanning functionality in OT, for people that has scanners with document feeder *like myself ;-)* (Feeders are not supported in GTS).
but some developer with knowledge on drivers would be necessary for that to happen.

As far as I know the original internal drivers "should work" on 32-bit build of OT, provided the code is not somehow disabled internally.
I've tried to use it in the past, with Morevna's 32 bit version of OT, but nothing happened.

**Comment by RodneyBaker** [comment]
I've been watching the (slow) progress of the TWAIN Working Group for some time.
I'll add the link here for reference:  https://www.twain.org/

There are a number of initiatives the working group has been advancing and some look quite promising.
However, the pace of progress has been very slow.

If/when the TWAIN projects are available that should (at least theoretically) all for scanner support via Opentoonz.

**Comment by RodneyBaker** [comment]
Labeling as bug due to nature of Opentoonz menu suggesting scanners can be used but with no reports of anyone actually being able to use scanners internally inside the program.  Further, documentation relates the process has been deprecated.

The desire going forward would certainly be for Opentoonz to have direct scanning capability.
It will be good to chart the way forward with consideration for both internal and external scanning capability while also including line testing and stop motion capability of webcam, DSLR, etc.  A special path also to GTS should be documented for those desiring to use that external capability.

A pressing question:  What to do with the menu items that are not working; specifically those related to internal scanning capability.

Recommend we chart a course for v1.7, v2.0 and beyond to fully explore short and long term options and potential.

**Comment by artisteacher** [comment]
It's a little funny to me that scanning support has been deprecated when OT supports traditional workflows. 

Personally I'd comment out the menu items until support has actually been added.

**Comment by gab3d** [comment]
Yes, I think the scanning related entries could, at least, be greyed-out in the menu.

In the long term, I'd love for OT to regain the scanning capabilities, especially because it could scan batches of drawings, using the scanner feeder capabilities, something GTS cannot currently perform.

**Comment by RodneyBaker** [comment]
Let's consider greying out these menu items for v1.8 if we cannot modify it to point to launching a third part application.

@shun-iwasawa You'll surely have a deeper insight into the importance of keeping the current menu as is or further justification for removing it.

**Comment by motoglaf** [comment]
I would recommend we re-add the scanning feature to Opentoonz. The reason why is GTS is not cross-platform and is only for Windows. Opentoonz serves Windows, Mac, and Linux. I think this feature is essential for how Opentoonz works with traditional animation. It seems very incomplete to have the ability to do things like pencil testing, cleanup, cel coloring, and etc without having the ability to scan in high-resolution frames and have them fed into your xsheet automatically such as how camera capture works. 

Camera capture can not be measured in the same way as it is for pencil testing and has limited resolution and cleanup options. Having a way to do clean consistent scans would be the best solution. Right now workflows that can't use GTS have to use a disjointed process of scanning and then importing and cleanup in Opentoonz. Why not have the scanning, cleanup, and other features all in one place? Think it makes sense for the workflow that Opentoonz offers.

**Comment by RodneyBaker** [comment]
Would love to see both Opentoonz and GTS move to Twain 2.4 if/when possible.

REF:  https://github.com/twain/twain-specification/tree/master/versions/2.4

**Comment by gab3d** [comment]
Anything scanning related in OpenToonz would be wonderful, of course.

**Comment by RodneyBaker** [comment]
I haven't heard much about the PDF/raster format in a very long time but that supposedly is something added to the v2.4 Twain header library.

REF:  https://pdfraster.org/

**Comment by RodneyBaker** [comment]
I'm actually getting some options in Opentoonz when I try to run scanning.
Never had that happen before.

![image](https://github.com/opentoonz/opentoonz/assets/2576432/42128c99-12c7-4b73-9802-62ea78a82d22)

**Comment by RodneyBaker** [comment]
Closing as a newer report has been opened that captures similar information.
That report references this report.
